boa_engine/src/context/mod.rs

```diff
@@ -162,14 +162,21 @@ impl Context {
         parser.parse_all(self)
     }
 
-    /// Parse the given source text in strict mode.
-    pub(crate) fn parse_strict<S>(&mut self, src: S) -> Result<StatementList, ParseError>
+    /// Parse the given source text with eval specific handling.
+    pub(crate) fn parse_eval<S>(
```

Comment:
We can probably use this in `boa_tester`, in the `$262` object that gets passed to the context. It has a special `eval` function that currently does the parsing twice, but can probably benefit from this.

Reply:
Ye, that's what I meant xD didn't remember it well. But yes, I get that. Maybe we can do the same somehow in `evalScript()`.